### PR TITLE
Add docs for BM provisioning and scaling

### PR DESCRIPTION
**Pull request by rabi**
Add content for BM provisioning, IPAM, scaling.

### CODE REVIEW

File: docs/scaling.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      nodes:
+      ...
+        edpm-compute-2:
+          hostName: edpm-compute-2
+      ...
+
+To deploy on the additional nodes, a new `OpenStackDataPlaneDeployment` CR should be
+created with the `OpenStackDataPlaneNodeSet` in the `nodeSets` section.
+
+Once deployment completes `OpenStackDataPlaneNodeSet` would be ready as shown below.
```
Comment (abays) [inline docs/scaling.md:53]: Wouldn't deployment only be complete after `OpenStackDataPlaneDeployment` is ready?

**Review by abays — COMMENTED** [review]
+1

This is great information!  Thank you for sharing @rabi.

**Review by fao89 — APPROVED** [review]

File: docs/scaling.md (hunk 1 above)
Comment (rabi) [reply docs/scaling.md:53]: Once `OpenStackDataPlaneDeployment` is Ready (i.e "SetupComplete" ),  we set the `OpenStackDataPlaneNodeSet` `DeploymentReadyCondition=True` [1] that would set it to `Ready` (i.e "NodeSet Ready").

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/controllers/openstackdataplanenodeset_controller.go#L303-L316

Comment (abays) [reply docs/scaling.md:53]: Ack

**Pull request merged**